File: rdk-sys/cpp/src/atom.cc
#include "rust/cxx.h"
#include <GraphMol/Atom.h>
#include <GraphMol/PeriodicTable.h>

namespace RDKit {

    std::shared_ptr<Atom> atom_from_symbol(const std::string &symbol) {
        Atom atom = Atom(symbol);
        return std::make_shared<Atom>(atom);
    }

    rust::String get_symbol(std::shared_ptr<Atom> atom) {
        return atom->getSymbol();
    }
}

File: rdk-sys/src/bindings/atom.rs
#[cxx::bridge(namespace = "RDKit")]
pub mod ffi {

    unsafe extern "C++" {
        include!("cpp/include/ro_mol.h");
        include!("cpp/include/atom.h");

        pub type Atom;
        pub type ROMol = crate::ro_mol_ffi::ROMol;

        pub fn atom_from_symbol(symbol: &CxxString) -> Result<SharedPtr<Atom>>;
        pub fn get_symbol(atom: SharedPtr<Atom>) -> String;
    }
}
